projects/baby_driver/test/test_spi_exchange.c

```diff
@@ -0,0 +1,354 @@
+  send_meta_data(0, 0, 7, 7, 0, 0, 1, 0x005b8d80);
+
+  send_data(7);
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL(8, s_num_messages);
+  test_equal_multiple_messages();
+
+  TEST_ASSERT_EQUAL(SPI_PORT_TEST_1, spi_port);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.baudrate, spi_settings.baudrate);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.mode, spi_settings.mode);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.mosi.port, spi_settings.mosi.port);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.mosi.pin, spi_settings.mosi.pin);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.miso.port, spi_settings.miso.port);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.miso.pin, spi_settings.miso.pin);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.sclk.port, spi_settings.sclk.port);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.sclk.pin, spi_settings.sclk.pin);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.cs.port, spi_settings.cs.port);
+  TEST_ASSERT_EQUAL(spi_settings_test_1.cs.pin, spi_settings.cs.pin);
+
+  TEST_ASSERT_EQUAL(1, s_callback_counter_status);
+  TEST_ASSERT_EQUAL(BABYDRIVER_MESSAGE_STATUS, s_received_data_status[0]);
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, s_received_data_status[1]);
+
+  clear_received_data_array();
+  s_num_messages = 0;
+
+  // second test using the default cs port and pin
+  send_meta_data(0, 0, 7, 7, 0, 0, 0, 0x005b8d80);
+
+  send_data(7);
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL(8, s_num_messages);
+  test_equal_multiple_messages();
+
+  TEST_ASSERT_EQUAL(2, s_callback_counter_status);
+  TEST_ASSERT_EQUAL(BABYDRIVER_MESSAGE_STATUS, s_received_data_status[0]);
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, s_received_data_status[1]);
+
+  clear_received_data_array();
+  s_num_messages = 0;
+
+  // third test for tx_len and rx_len edge cases (0)
+  send_meta_data(0, 0, 0, 0, 0, 0, 0, 0x005b8d80);
+
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL(0, s_num_messages);
+
+  TEST_ASSERT_EQUAL(3, s_callback_counter_status);
+  TEST_ASSERT_EQUAL(BABYDRIVER_MESSAGE_STATUS, s_received_data_status[0]);
+  TEST_ASSERT_EQUAL(STATUS_CODE_OK, s_received_data_status[1]);
+
+  clear_received_data_array();
+  s_num_messages = 0;
+
+  /*
+    // fourth test for tx_len and rx_len edge cases (255)
+    send_meta_data(0, 0, 255, 255, 0, 0, 0, 0x005b8d80);
+
+    for (uint8_t i = 0; i < 36; i++) {
+      send_data(7);
+    }
+    send_data(3);
+    for(uint8_t i = 0; i < 37; i++){
+      MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+    }
+    // right now there is no way to receive this many messages without the helper issue
+
+    TEST_ASSERT_EQUAL(292, s_num_messages);
+    test_equal_multiple_messages();
+
+    clear_received_data_array();
+    s_num_messages = 0;
+  */
```

Comment:
I think it's alright to omit this test with just a comment saying what it is and why it was omitted.